src/hosted-buttons/index.js

```diff
@@ -34,24 +39,59 @@ export const getHostedButtonsComponent = (): HostedButtonsComponent => {
         selector,
       });
 
-      // $FlowFixMe
-      Buttons({
-        hostedButtonId,
-        style,
-        onInit,
-        onClick,
-        createOrder: buildHostedButtonCreateOrder({
-          enableDPoP,
-          hostedButtonId,
-          merchantId,
-        }),
-        onApprove: buildHostedButtonOnApprove({
-          enableDPoP,
+      if (shouldRenderSDKButtons(fundingSources)) {
+        const { flexDirection } = getFlexDirection({ ...style });
+
+        buildButtonContainer({ flexDirection, selector });
+
+        // Only render 2 buttons max
+        fundingSources?.slice(0, 2).forEach((fundingSource, index) => {
+          // $FlowFixMe
+          Buttons({
+            hostedButtonId,
+            fundingSource,
+            style: {
+              ...style,
+              color: getButtonColor(style.color, fundingSource),
+            },
+            onInit,
+            onClick,
+            createOrder: buildHostedButtonCreateOrder({
+              enableDPoP,
+              hostedButtonId,
+              merchantId,
+            }),
+            onApprove: buildHostedButtonOnApprove({
+              enableDPoP,
+              hostedButtonId,
+              merchantId,
+            }),
+          }).render(
```

Comment:
we'd still want to check `Button.isEligible()` before rendering. Venmo for instance has a lot of client-only eligibility checks

Reply:
Updated in this [commit](https://github.com/paypal/paypal-checkout-components/pull/2375/commits/ac9f189a7d1d6e145742691a1707958bd7908fa1)